modules/gateway/nodes.go

```diff
@@ -142,28 +142,35 @@ func (g *Gateway) permanentNodePurger(closeChan chan struct{}) {
 		// Try connecting to the random node. If the node is not reachable,
 		// remove them from the node list. Make sure that the dial is stopped
 		// early if the gateway is shutting down.
-		cancelDialChan := make(chan struct{})
-		dialDoneChan := make(chan struct{})
-		defer close(dialDoneChan)
-		dialer := &net.Dialer{
-			Timeout: dialTimeout,
-			Cancel:  cancelDialChan,
-		}
-		go func() {
-			select {
-			case <-cancelDialChan:
-			case <-g.threads.StopChan():
-				close(cancelDialChan)
+		var conn net.Conn
+		func() {
```

Comment:
Can you elaborate on the addition on the anonymous function? The only difference I see is where `close(dialDoneChan)` is called.